test/test_autograd.py

```diff
@@ -1551,7 +1535,7 @@ def backward(ctx, grad_output):
 
         x = Variable(torch.randn(2, 2), requires_grad=True)
         out = Reenter.apply(x)
-        out.sum().backward()
+        out.sum().backward(create_graph=True)
```

Comment:
The line numbers have changed, but I now see what's the problem. Can we add `torch.with_grad` so that such ops can ensure that they always run in a differentiable context, even when we don't instantiate the graph for backward (which shouldn't be necessary)